examples/fixture/asm/r1cs/not_equal.asm

```diff
@@ -0,0 +1,12 @@
+@ noname.0.7.0
+
+v_4 == (v_1 + -1 * v_2) * (v_3)
+v_5 + -1 == (-1 * v_4) * (1)
+v_6 == (v_1 + -1 * v_2) * (v_5)
+0 == (v_6) * (1)
+1 == (-1 * v_5 + 1) * (1)
+v_8 == (v_1 + -7) * (v_7)
+v_9 + -1 == (-1 * v_8) * (1)
+v_10 == (v_1 + -7) * (v_9)
+0 == (v_10) * (1)
+1 == (-1 * v_9 + 1) * (1)
```

Comment:
so many constraints :o I'm wondering if we're missing good optimizations here

Reply:
i agree there definitely can be some optimization, i'm imagining in the `is_zero_circuit`?